New contribution guide and philosophy document (#1793)
lemonade-sdk/lemonade: docs/dev/contribute.md

# Contributing to Lemonade

> We're excited that you are interesting in contributing!

Reviewer: ```suggestion
We're excited that you are interested in contributing!
```

Please carefully review Lemonade's [philosophy](./philosophy.md) before making a pull request. As a contributor, you can anticipate the vast majority of reviewer feedback by aligning your design to the philosophy.

## Development Process

### Merging a Contribution

> 1. We highly recommend that all contributors join the [Lemonade discord community](https://discord.gg/5xXzkMu8Zk) meet [the maintainers](#maintainers) and get a sense of what is trending.

Reviewer 2: ```suggestion
1. We highly recommend that all contributors join the [Lemonade discord community](https://discord.gg/5xXzkMu8Zk), meet [the maintainers](#maintainers) and get a sense of what is trending.
```

2. Depending on the complexity of your contribution:
    - Simple fixes: just make a PR.
    - Features: contact [a maintainer](#maintainers) who works in the area of your feature and get them to agree to review it before you start coding.
    - Architectural changes: write an issue explaining the justification and design and bring it to the Discord for debate.
3. Create a fork of Lemonade repo, implement your code, and then make a pull request to merge your code back to the main repo. Assign the reviewer(s) you discussed the change with.

### Picking a Project

Not sure what to work on? Come to the feature-requests and troubleshooting channels on the Discord and see what people need!

### Issues

> Issues are a great way to document a bug or feature request. However, Lemonade is a community-driven project and you still need to find someone to implement your issue. It is highly recommended that you bring your issue to the [Lemonade discord community](https://discord.gg/5xXzkMu8Zk) and connect with a contributor who is wants to implement it.

Reviewer: ```suggestion
Issues are a great way to document a bug or feature request. However, Lemonade is a community-driven project and you still need to find someone to implement your issue. It is highly recommended that you bring your issue to the [Lemonade discord community](https://discord.gg/5xXzkMu8Zk) and connect with a contributor who wants to implement it.
```

### Reviewer Expectation

Each contribution needs to:

1. Adhere to the [philosophy](./philosophy.md).

> 2. Pass the CI tests. Make sure to check the code *before* allowing CI to run!

Reviewer 3: ```suggestion
2. Pass the CI tests. Make sure to check the code, and compile locally *before* allowing CI to run!
```

3. Sustain the overall code quality and standards of the rest of the repo.

### AI Policy

Contributors are encouraged to use AI to code their project. However, please review your AI's code yourself before asking another human to review it.

> Reviewers are expected to use tools like `claude /review` to save time by analyzing code quality and edge cases. If you have AI tools make PR comments please clearly identify which comments are written by AI and which are written by you.

Reviewer: ```suggestion
Reviewers are expected to use tools like `claude-code /review` to save time by analyzing code quality and edge cases. If you have AI tools make PR comments please clearly identify which comments are written by AI and which are written by you.
```

> Reviewers are expected to use tools like `claude /review` to save time by analyzing code quality and edge cases. If you have AI tools make PR comments please clearly identify which comments are written by AI and which are written by you.

Reviewer 2: ```suggestion
Reviewers are expected to use tools like `claude-code /review` to save time by analyzing code quality and edge cases. If AI tools are used to generate PR comments, please clearly identify which comments are AI-generated and which are authored by you
```

__Please do not use AI to write issues__. If you feel an issue is important enough for a human to read it then please take the time to write it yourself.

## Maintainers

While each maintainer is welcome to work on any part of the Lemonade codebase, each maintainer does have specific knowledge of certain areas. You should use their knowledge as a starting point for designing your contribution, and they will be the ones to review your contribution when it is ready.

### Features

| Subject area | Maintainers |
| --- | --- |

> | GUI | Primary maintainer needed, @danielholanda, @sofiageo, @jeremyfowers |

Reviewer 4: Happy to volunteer more of my time here. Even happier if a community member can step up as a core maintainer.

Reviewer 3: There should be a difference from GUI and Backend as far as core goes. Those use to be two roles before full-stack, and I still believe are two different things when talking about JS/HTML/CSS vs C++

| CLI | @bitgamma (discord: mikkoph), @jeremyfowers |
| Networking (HTTP) | @Geramy, @bitgamma (discord: mikkoph) |

> | Mobile Apps | @Geramy |

Reviewer 3: We should probably add a core contributors area too, basically the backend which I assume is a lot of us actually, but its the catch all for when one of the things is not here like engines for example. I would prefer to know when someone that isn't a maintainer is making adjustments to the backend so I can be aware and not lag behind in major developments that are so important as that.

Author: I haven't been making a big distinction between maintainers and core contributors. I've just been graduating people to maintainer pretty fast. Let's take another look at this as we scale more and people sort themselves into clear tiers?

| Coding Agents | @sawansri |
| Website | @jeremyfowers |
| Content and guides | @vgodsoe |
| LemonAIde bot | @kpoineal |

### AI Support

| Subject area | Maintainers |
| --- | --- |
| LLMs | @sawansri, @bitgamma (discord: mikkoph), @ramkrishna2910 |
| Speech AI | @bitgamma (discord: mikkoph), @ramkrishna2910 |
| Image AI | @bitgamma (discord: mikkoph), @kenvandine, @ramkrishna2910 |

### OS Support

| Subject area | Maintainers |
| --- | --- |
| Linux | @superm1, @kenvandine, @bitgamma (discord: mikkoph), @sofiageo |
| Ubuntu, Debian, and Snap | @kenvandine, @superm1 |
| Arch | @sofiageo, @bitgamma (discord: mikkoph) |
| Fedora | Maintainer needed |
| macOS | @Geramy, @ramkrishna2910 |
| Windows | @kpoineal, @jeremyfowers |
| Docker | @siavashhub (discord: icmpreply) |

### Backend Support

| Subject area | Maintainers |
| --- | --- |
| Adding new backends | @ramkrishna2910, @bitgamma (discord: mikkoph), @jeremyfowers |
| AMD ROCm | @superm1, @Geramy, @danielholanda |
| AMD NPU | @ramkrishna2910, @jeremyfowers |

> | Intel | Maintainer Needed |

Reviewer 3: ```suggestion
| Intel | Primary Maintainer Needed | @Geramy 
```
